gems/smithy/spec/spec_helper.rb

```diff
@@ -23,18 +23,26 @@ class << self
     def generate(modules, type, options = {})
       model = load_model(modules, options)
       plan = create_plan(modules, model, type, options)
+      sdk_dir = plan.options[:destination_root]
       smith(plan)
 
-      $LOAD_PATH << ("#{plan.options[:destination_root]}/lib")
+      $LOAD_PATH << ("#{sdk_dir}/lib")
       require "#{plan.options[:gem_name]}#{type == :types ? '-types' : ''}"
-      plan.options[:destination_root]
+
+      setup_rbs_spytest(modules, sdk_dir) if options.fetch(:rbs_test, ENV.fetch('SMITHY_RUBY_RBS_TEST', nil))
+      sdk_dir
+    rescue StandardError => e
```

Comment:
Does the after{} block not already do this? When does this error happen?

Reply:
When an exception is raised, the method does not return a value, meaning that the `@tmpdir` in specs does not get set.  The `after` block does get called, but it gets called with `nil` for directory.  This ensures that the directory is cleaned up correctly.

These errors can happen for a few reasons:
1. errors during code generation.
2. errors requiring generated code.
3. invalid rbs (causes loading the rbs env to fail).

With the changes I've made we now get nice clean error messages/stack traces and no tmp dirs are left around!